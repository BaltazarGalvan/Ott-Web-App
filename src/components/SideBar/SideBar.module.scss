@use '../../styles/variables';
@use '../../styles/theme';

//
// jwSidebarBackdrop
// --------------------------------

.backdrop {
  position: fixed;
  top: 0;
  left: 0;
  z-index: variables.$sidebar-z-index - 1;
  display: none;

  width: 100%;
  height: 100%;

  background: rgba(variables.$black, 0.4);

  &.visible {
    display: block;
  }
}

//
// jwSidebar
// --------------------------------

.sidebar {
  position: fixed;
  top: 0;
  z-index: variables.$sidebar-z-index;

  width: 270px;
  max-width: 90vw;
  height: 100vh;
  max-height: 100vh;

  overflow-x: hidden;
  overflow-y: auto;
  background-color: var(--body-background-color);
  transform: translateX(-100%);

  visibility: hidden;
  transition: transform 0.2s cubic-bezier(0.52, 0.51, 0.2, 1);

  &.open {
    transform: translateX(0);
    visibility: visible;
  }
}

.group {
  display: flex;
  flex-direction: column;
  padding: variables.$base-spacing;
}

.divider {
  width: 270px;
  max-width: 90vw;
  border-top: 1px solid rgba(variables.$white, 0.12);
  opacity: 0.12;
}
