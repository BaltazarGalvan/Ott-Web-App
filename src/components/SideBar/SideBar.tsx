import React, { Fragment } from 'react';
import classNames from 'classnames';

import Close from '../../icons/Close';
import ButtonLink from '../ButtonLink/ButtonLink';

import styles from './SideBar.module.scss';

type SideBarProps = {
  isOpen: boolean;
  onClose: () => void;
  playlistMenuItems: JSX.Element[];
};

const SideBar: React.FC<SideBarProps> = ({ isOpen, onClose, playlistMenuItems }) => {
  return (
    <Fragment>
      <div
        className={classNames(styles.backdrop, {
          [styles.visible]: isOpen,
        })}
        onClick={onClose}
      />
      <div
        className={classNames(styles.sidebar, {
          [styles.open]: isOpen,
        })}
        onClick={onClose}
      >
        <div className={styles.group} aria-label="close menu" role="button">
          <Close />
        </div>
        <nav className={styles.group}>
          <ButtonLink label="Home" to="/" />
          {playlistMenuItems}
          <hr className={styles.divider} />
          <ButtonLink label="Settings" to="/" />
        </nav>
      </div>
    </Fragment>
  );
};

export default SideBar;
