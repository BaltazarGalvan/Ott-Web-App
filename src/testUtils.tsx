import { BrowserRouter as Router } from 'react-router-dom';
import React, { ReactNode, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

interface WrapperProps {
  children?: ReactNode;
}

export const wrapper = ({ children }: WrapperProps) => <Router>{children}</Router>;

const customRender = (ui: ReactElement, options?: RenderOptions) => render(ui, { wrapper, ...options });

export const mockWindowLocation = (path: string) => {
  Object.defineProperty(window, 'location', {
    value: {
      pathname: path,
      assign: jest.fn(),
    },
  });
};

export const mockMatchMedia = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    })),
  });
};

export { customRender as render };
