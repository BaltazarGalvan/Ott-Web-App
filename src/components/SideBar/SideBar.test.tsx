import React from 'react';

import { render } from '../../testUtils';

import SideBar from './SideBar';

describe('<SideBar />', () => {
  test('renders sideBar', () => {
    const { container } = render(
      <SideBar sideBarOpen={true} closeSideBar={jest.fn()} />,
    );

    expect(container).toMatchSnapshot();
  });
});
