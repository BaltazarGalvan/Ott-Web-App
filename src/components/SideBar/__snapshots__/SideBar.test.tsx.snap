// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> renders no sideBar 1`] = `
<div>
  <div
    class=""
  >
    <div
      class=""
    >
      <nav>
        <svg
          class=""
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </nav>
      <div>
        <a
          aria-current="page"
          class="function link() { [native code] } active"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
        <a
          aria-current="page"
          class="function link() { [native code] } active"
          href="/"
        >
          <span>
            Test
          </span>
        </a>
        <hr />
        <a
          aria-current="page"
          class="function link() { [native code] } active"
          href="/"
        >
          <span>
            Settings
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<SideBar /> renders sideBar 1`] = `
<div>
  <div
    class="undefined"
  >
    <div
      class="undefined"
    >
      <nav>
        <svg
          class=""
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </nav>
      <div>
        <a
          aria-current="page"
          class="function link() { [native code] } active"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
        <a
          aria-current="page"
          class="function link() { [native code] } active"
          href="/"
        >
          <span>
            Test
          </span>
        </a>
        <hr />
        <a
          aria-current="page"
          class="function link() { [native code] } active"
          href="/"
        >
          <span>
            Settings
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
